harness/src/search/search.ts

```diff
@@ -0,0 +1,233 @@
+    }
+
+    console.log(JSON.stringify(data, null, 2));
+    console.log("Remaining iterations:", blocksToProcess.length - i - 1);
+  }
+
+  // Write output file
+  const outfile = `${chainId} ${data.blockRange.start}-${data.blockRange.end}.json`;
+  fs.mkdirSync(outputPath, { recursive: true });
+  fs.writeFileSync(path.join(outputPath, outfile), JSON.stringify(data, null, 2));
+}
+
+function pushArrayUpto<T>(arr: T[], itm: T, size: number = 32) {
+  if (arr.length === size) {
+    return;
+  }
+  arr.push(itm);
+}
+
+async function parseAccount(provider: JsonRpcProvider, tx: any, idx: number, ignoreAddrs: string[]): Promise<string[]> {
+  const accounts = [tx.from, tx.to];
+  let contracts: string[] = [];
+  for await (const account of accounts) {
+    if (!account) {
+      continue;
+    }
+    // Check if account matches any part of the addresses in IGNORE_ADDRS_MATCH
+    const shouldIgnore = ignoreAddrs.some(ignoreAddr => account.includes(ignoreAddr));
+    if (shouldIgnore) {
+      continue; // Skip this account if it matches any ignore address pattern
+    }
+    const code = await provider.getCode(account);
+    if (code === '0x') {
+      if (Number(tx.transactionIndex) === idx) {
+        pushArrayUpto(data.account.eoa, {
+          blockNumber: Number(tx.blockNumber),
+          address: account
+        });
+      }
+    } else {
+      if (Number(tx.transactionIndex) === idx) {
+        pushArrayUpto(data.account.contract, {
+          blockNumber: Number(tx.blockNumber),
+          address: account
+        });
+      }
+      contracts.push(account);
+    }
+  }
+  return contracts;
+}
+
+async function parseStorage(provider: JsonRpcProvider, contracts: string[], blockNumber: string): Promise<boolean> {
+  if (contracts.length === 0) {
+    return false;
+  }
+  for await (const contract of contracts) {
+    // Random slot from 0 to 10
+    const randomSlot = Math.floor(Math.random() * 11);
+    const storage = await provider.getStorage(contract, randomSlot, blockNumber);
+    if (storage !== bytes32(0)) {
+      pushArrayUpto(data.storage.nonzero, {
+        blockNumber: Number(blockNumber),
+        address: contract,
+        slot: randomSlot,
+      });
+      return true;
+    }
+  }
+  return false;
+}
+
+async function parseTx(tx: any) {
+  const type = Number(tx.type).toString() as keyof typeof data.tx.type;
+  const txId: TxId = {
+    hash: tx.hash,
+    blockNumber: Number(tx.blockNumber),
+    txIdx: Number(tx.transactionIndex),
+  };
+  if (!(type === "0" || type === "1" || type === "2" || type === "3")) {
+    return;
+  }
+  pushArrayUpto(data.tx.type[type], txId);
+  const numBytesTxData = getNumBytes(tx.input);
+  // if (tx.accessList !== undefined && tx.accessList !== null && tx.accessList.length !== 0) {
+  //   return;
+  // }
+  const accessListRlp = objectToRlp(tx.accessList ?? {});
+  const aclNumBytesRlp = getNumBytes(accessListRlp);
+  if (numBytesTxData <= 8192 && aclNumBytesRlp <= 800) {
```

Comment:
is there any way to use the `QueryBuilder`'s categorization for this stuff? otherwise we have to. maintain two copies of the same logic

Reply:
I'll leave a `// TODO:` for the QueryBuilder PR but I'd prefer to not add the sdk-core dependency in there for now